benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/configuration/goal/explorationspace/workload/users/UserValuesYamlProtocol.scala

```diff
@@ -0,0 +1,162 @@
+import cloud.benchflow.dsl.definition.configuration.goal.explorationspace.step.RangeWithStep
+import cloud.benchflow.dsl.definition.errorhandling.BenchFlowDeserializationException
+import cloud.benchflow.dsl.definition.errorhandling.YamlErrorHandler.{ deserializationHandler, unsupportedReadOperation, unsupportedWriteOperation }
+import net.jcazevedo.moultingyaml.{ DefaultYamlProtocol, YamlFormat, YamlObject, YamlString, YamlValue, _ }
+
+import scala.util.{ Failure, Success, Try }
+
+/**
+ * @author Jesper Findahl (jesper.findahl@gmail.com)
+ *         created on 2017-05-23
+ */
+object UserValuesYamlProtocol extends DefaultYamlProtocol {
+
+  val ValuesKey = YamlString("values")
+  val RangeKey = YamlString("range")
+  val StepKey = YamlString("step")
+
+  val Level = s"${ExplorationSpaceYamlProtocol.Level}.(some int dimension)"
+
+  private def keyString(key: YamlString) = s"$Level.${key.value}"
+
+  implicit object UserValuesReadFormat extends YamlFormat[Try[UserValues]] {
+
+    override def read(yaml: YamlValue): Try[UserValues] = {
+
+      val yamlObject = yaml.asYamlObject
+
+      // all options stored as a list of values for easier use later on
+
+      for {
+        values <- deserializationHandler(
+          yamlObject match {
+
+            // handle values specification
+            case _ if yamlObject.getFields(ValuesKey).nonEmpty =>
+              assertValidValues(
+                yamlObject.getFields(ValuesKey).headOption.map(_.convertTo[List[Int]]).get).get
+
+            // handle step and range specification
+            case _ if yamlObject.getFields(StepKey).nonEmpty && yamlObject.getFields(RangeKey).nonEmpty =>
+              getStepList(
+                yamlObject.getFields(RangeKey).headOption.map(_.convertTo[List[Int]]).get,
+                yamlObject.getFields(StepKey).headOption.map(_.convertTo[String]).get).get
+
+            // handle range only specification
+            case _ if yamlObject.getFields(RangeKey).nonEmpty =>
+              getRangeList(
+                yamlObject.getFields(RangeKey).headOption.map(_.convertTo[List[Int]]).get).get
+
+          },
+          keyString(ValuesKey))
+
+      } yield UserValues(values = values)
+
+    }
+
+    override def write(obj: Try[UserValues]): YamlValue = unsupportedWriteOperation
+  }
+
+  implicit object UserValuesWriteFormat extends YamlFormat[UserValues] {
+
+    override def write(obj: UserValues): YamlValue = YamlObject {
+
+      Map[YamlValue, YamlValue](
+        ValuesKey -> obj.values.toYaml)
+
+    }
+
+    override def read(yaml: YamlValue): UserValues = unsupportedReadOperation
+
+  }
+
+  private def assertValidValues(values: List[Int]): Try[List[Int]] = {
+
+    // check that all values are in increasing order
+    val increasingOrder = values.sliding(2).forall {
+      case List(x, y) => x < y
+    }
+
+    if (!increasingOrder) {
+      Failure(new BenchFlowDeserializationException("values must be increasing", new Throwable))
```

Comment:
When you report this `Failure` is it clear in which part of the DSL definition there was the issue?